docs/src/examples/redis-clustering-aware.md

```diff
@@ -0,0 +1,117 @@
+# Redis Clustering with cluster aware client
+
+The following guide shows you how to configure Shotover to support proxying Redis cluster *aware* clients to [Redis cluster](https://redis.io/topics/cluster-spec).
+
+## Overview
+
+In this example, we will be connecting to a Redis cluster that has the following topology:
+* `172.16.1.2:6379`
+* `172.16.1.3:6379`
+* `172.16.1.4:6379`
+* `172.16.1.5:6379`
+* `172.16.1.6:6379`
+* `172.16.1.7:6379`
+
+Shotover will be deployed as a sidecar to each node in the Redis cluster, listening on `6380`. Use the following [docker-compose.yml](https://github.com/shotover/shotover-proxy/blob/main/shotover-proxy/examples/redis-cluster-ports-rewrite/docker-compose.yml) to run the Redis cluster and Shotover sidecars.
+
+```console
+curl -L https://raw.githubusercontent.com/shotover/shotover-proxy/main/shotover-proxy/examples/redis-cluster-ports-rewrite/docker-compose.yml --output docker-compose.yml
+docker-compose -f docker-compose.yml up
+```
+
+Below we can see an example of a Redis node and it's Shotover sidecar. Notice they are running on the same network address (`172.16.1.2`) and the present directory is being mounted to allow Shotover to access the config and topology files.
+
+```YAML
+
+redis-node-0:
+  image: docker.io/bitnami/redis-cluster:6.0-debian-10
+  networks:
+  cluster_subnet:
+    ipv4_address: 172.16.1.2
+  volumes:
+    - redis-cluster_data-0:/bitnami/redis/data
+  environment:
+    - 'ALLOW_EMPTY_PASSWORD=yes'
+    - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2'
+
+shotover-0:
+  restart: always
+  depends_on:
+    - redis-node-0
+  image: shotover/shotover-proxy
+  network_mode: "service:redis-node-0"
+  volumes:
+    - type: bind
+      source: $PWD
+      target: /config
+
+```
+
+In this example we will use `redis-benchmark` with cluster mode enabled as our Redis cluster aware client application. 
+
+## Configuration
+
+First we will modify our `topology.yaml` file to have a single Redis source. This will:
+
+* Define how Shotover listens for incoming connection from our client application (`redis-benchmark`).
+* Configure Shotover to connect to the Redis node via our defined contact point.
+* Configure Shotover to rewrite all Redis ports with our Shotover port when the cluster aware driver is talking to the cluster, through Shotover.
+* Connect our Redis Source to our Redis cluster sink (transform).
+
+```yaml
+{{#include ../../../shotover-proxy/examples/redis-cluster-ports-rewrite/topology.yaml}}
+```
+
+Modify an existing `topology.yaml` or create a new one and place the above example as the files contents.
+
+You will also need a [config.yaml](https://raw.githubusercontent.com/shotover/shotover-proxy/main/shotover-proxy/config/config.yaml) to run Shotover.
+
+```console
+curl -L https://raw.githubusercontent.com/shotover/shotover-proxy/main/shotover-proxy/config/config.yaml --output config.yaml
```

Comment:
this should point at the examples config file rather than the generic config file.

Reply:
Addressed this in later commit